client_test.go

```diff
@@ -0,0 +1,85 @@
+package amqp
```

Comment:
I'd prefer to handle connection setup testing in the integration tests. Please remove this and the other supporting changes.